lib/ansible_tower_client/base_models/credential.rb

```diff
@@ -1,7 +1,43 @@
 module AnsibleTowerClient
   class Credential < BaseModel
     def override_raw_attributes
-      { :organization => :organization_id }
+      {
+        :organization    => :organization_id,
+        :credential_type => :credential_type_id,
+      }
+    end
+
+    # https://github.com/ansible/awx/blob/1328fb80a02ef4e37bc021eb07d4be041a41f937/awx/main/models/credential/__init__.py#L76-L90
+    KIND_CHOICES = {
+      'ssh'        => 'Machine',
+      'net'        => 'Network',
+      'scm'        => 'Source Control',
+      'aws'        => 'Amazon Web Services',
+      'vmware'     => 'VMware vCenter',
+      'satellite6' => 'Red Hat Satellite 6',
+      'cloudforms' => 'Red Hat CloudForms',
+      'gce'        => 'Google Compute Engine',
+      'azure_rm'   => 'Microsoft Azure Resource Manager',
+      'openstack'  => 'OpenStack',
+      'rhv'        => 'Red Hat Virtualization',
+      'insights'   => 'Insights',
+      'tower'      => 'Ansible Tower',
+    }.invert.freeze
+
+    # https://github.com/ansible/awx/blob/1328fb80a02ef4e37bc021eb07d4be041a41f937/awx/main/models/credential/__init__.py#L301
+    def kind
+      @data['kind'] ||= begin
+        kind = credential_type.kind
+        kind == 'cloud' && KIND_CHOICES[credential_type.name] || kind
```

Comment:
Prefer ternary operator for readability

Reply:
Ternary do not suit here well, so I had to add default value to `KIND_CHOICES` hash few lines above, now logic works the same as it was expected. Done.